docs/metrics-yaml-format.md

```diff
@@ -0,0 +1,254 @@
+         omitted in the definition, the agent aggregates the metric on the omitted tag using the aggregation function 
+         based on `type`. Counters are aggregated by `SUM` and gauges by `AVG`. This could be overridden by using this field.
+         Valid values are `SUM`, `AVG`, `MAX`, `MIN`, `DISCARD`.
+        * `unit`: Unit of measurement for this metric e.g. `ms`, `bytes`, etc.
+    * Each `tag` can have name and value fields:
+        * `name`: Name of the tag. Make sure the tag name is unique. Prefix metric namespace to the tag name.  
+        * `value`: Reference to the name from where this tag has to be extracted. This could be a metric name defined 
+        under the observation or the placeholder in `path` or `objectName`. For metric, use `eval` function
+        to refer to metric. You can also use [built-in functions](./built-in-functions.md) to modify the values before ]
+        sending to output.
+
+## Metric Data Types
+
+Sematext App Agent supports the following metric data types:
+
+* `gauge`, `long_gauge`: Gauge data type. Default aggregation is `AVG`
+* `counter`, `long_counter`: Counter data type. Default aggregation is `SUM` 
+* `text`: Textual data type. Typically used for metrics that have to be extracted as tags
+
+## Derived Metrics
+
+Derived metrics are calculated by applying built-in or custom functions on other metrics. Derived metrics can be used to
+manipulate the values of metrics before sending to the receiver. Some of the use cases are to extract a number from a string 
+value or derive a metric as a function of 2 attributes. 
+
+For derived metrics, the `source` field of the metric has to be of the format: `func:<function-name>(<params>)`. 
+
+* `func`: Denotes that metric has to be derived by applying the function specified.
+* `<function-name>`: Function to invoked. In the case of [built-in functions](./built-in-functions.md), 
+just function name is enough. For custom functions, specify the function name with fully qualified classname. e.g.
+`func:com.sematext.spm.client.solr.CalculateWarmupTime(warmupTime,searcherName)`. The function class should extend 
+`com.sematext.spm.client.observation.CalculationFunction`
+* `<params>`: Optional params to function. All params are of type String and will be interpreted and converted 
+accordingly in the function implementation.
+
+By default, the result of the function will be interpreted as a gauge. If the result is counter, then prepend the function
+reference with `as_counter`. For example,
+
+```yaml
+  - name: cache.lookups
+    source: as_counter:func:ExtractLongFromMapString(Value,cumulative_lookups)
+    type: counter
+    label: cache lookups
+    description: lookups count
+    stateful: true
+```
+
+Some of the built-in functions can be applied to tags also. 
+
+## Percentiles
+
+Sematext App Agent, can automatically calculate percentiles for a metric and send to the receiver. The percentiles to be
```

Comment:
Sematext App Agent, can => Sematext App Agent can 
for a metric and send to the receiver => for a metric and send it